Source/JavaScriptCore/builtins/JSIteratorConstructor.js

```diff
@@ -71,3 +71,44 @@ function from(value)
     // 6. Return wrapper.
     return @wrapForValidIteratorCreate(iterator, iteratorNextMethod);
 }
+
+// https://tc39.es/proposal-iterator-sequencing/#sec-iterator.concat
+function concat(/* ...items */)
+{
+    "use strict";
+    var len = @argumentCount();
+
+    var iterables = [];
+    for (var i = 0; i < @argumentCount(); ++i) {
+        var iterable = arguments[i];
+        if (!@isObject(iterable))
+            @throwTypeError("Iterator.concat expects all arguments to be objects");
+
+        var openMethod = iterable.@@iterator;
+        if (@isUndefinedOrNull(openMethod))
+            @throwTypeError("Iterator.concat expects all arguments to be iterable");
+
+        @putByValDirect(iterables, i, {openMethod, iterable});
+    }
+
+    var generator = (function*() {
+        for (var iterable of iterables) {
```

Comment:
This is observable, please use `for`. Also I would consider using two arrays rather than array of objects.